pages/[metricType]/[metricId].tsx

```diff
@@ -0,0 +1,104 @@
+import { format } from 'date-fns'
+import ReactECharts from 'echarts-for-react'
+import { GraphQLClient } from 'graphql-request'
+import Head from 'next/head'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+
+import { DateRangePicker, Layout } from '@/components'
+import { CounterQuery, TimeSeriesQuery } from '@/graphql'
+import { buildCounterChartConfig, buildTimeSeriesChartConfig } from '@/utils'
+
+const client = new GraphQLClient(
+  process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT_US_EAST_2
+)
+
+export default function MetricVisualization() {
+  const [title, setTitle] = useState('')
+  const [description, setDescription] = useState('')
+  const [options, setOptions] = useState('' as any)
+  const [startDate, setStartDate] = useState(new Date('2021-01-01T10:00:00Z'))
+  const [stopDate, setStopDate] = useState(new Date('2022-05-21T10:00:00Z'))
+
+  const router = useRouter()
+  const { metricId, metricType } = router.query
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const accessToken = window.localStorage.getItem('accessToken')
+        client.setHeader('authorization', 'Bearer ' + accessToken)
+
+        const start = format(startDate, 'yyyy-MM-dd')
+        const stop = format(stopDate, 'yyyy-MM-dd')
+
+        let response = {} as any
+        // ReturnType<typeof f>
```

Comment:
I created another follow-up ticket to add GraphQL type generation. I didn't want to mock the types out too much, since they come from our schema.